### PR TITLE
Add a test-example that runs on Android

### PR DESCRIPTION
**Pull request by MarijnS95**
Depends on #20

Run with:

	cargo apk r --example android

(Requires `cargo-apk` and an Android SDK+NDK to be installed)

Outputs:

	I app_dirs2: android: UserConfig: Ok("/data/user/0/rust.example.android/unnecessary")
	I app_dirs2: android: UserData: Ok("/data/user/0/rust.example.android/files/unnecessary")
	I app_dirs2: android: UserCache: Ok("/data/user/0/rust.example.android/cache/unnecessary")
	I app_dirs2: android: SharedConfig: Ok("/storage/emulated/0/Android/data/rust.example.android/files/unnecessary")
	I app_dirs2: android: SharedData: Ok("/storage/emulated/0/Android/data/rust.example.android/files/unnecessary")

### CODE REVIEW

File: examples/android.rs [hunk 1]
```diff
@@ -0,0 +1,24 @@
+use app_dirs2::*;
+use log::info;
+
+const NDK_APP_INFO: AppInfo = AppInfo {
+    name: "unnecessary",
```
Comment (MarijnS95) [inline examples/android.rs:5]: I'm also creating this example to see if anyone else feels like adding a `NO_NAME` similar to `NO_AUTHOR`. On Android all these directories already contain the app/package name inside, and have thus little use for an extra name in the path, as shown in the PR message.

File: src/imp/platform/android.rs [hunk 2]
```diff
@@ -47,11 +57,13 @@ pub fn get_app_dir(t: AppDataType) -> Result<PathBuf, AppDirsError> {
...
-        AppDataType::UserConfig => get_jni_app_dir(&context, &env, "getDataDir")?,
-        AppDataType::UserData => get_jni_app_dir(&context, &env, "getFilesDir")?,
-        AppDataType::UserCache => get_jni_app_dir(&context, &env, "getCacheDir")?,
-        AppDataType::SharedData => get_jni_app_dir(&context, &env, "getExternalFilesDir")?, // Deprecated in Android 11+
-        AppDataType::SharedConfig => get_jni_app_dir(&context, &env, "getExternalFilesDir")?, // Deprecated in Android 11+
+        AppDataType::UserConfig => get_jni_app_dir(&context, &env, "getDataDir", false)?,
+        AppDataType::UserData => get_jni_app_dir(&context, &env, "getFilesDir", false)?,
+        AppDataType::UserCache => get_jni_app_dir(&context, &env, "getCacheDir", false)?,
+        AppDataType::SharedData | AppDataType::SharedConfig => {
+            get_jni_app_dir(&context, &env, "getExternalFilesDir", true)?
```
Comment (kornelski) [inline src/imp/platform/android.rs:64]: Why this change? Didn't it work before?

**Comment by kornelski** [comment]
Thanks
